### PR TITLE
✨ Fancier input validation for entering a read-only wallet

### LINKED ISSUE #557: [Bug]: Funky error states @ on read-only form
### Discord Discussion Link

_No response_

### What browsers are you seeing the problem on?

Chrome

### What were you trying to do?

Enter an incorrect ETH address and get error messaging inside the extension that I need to supply a properly-formatted eth address

### What did not work?

I get an OS pop-up that I need to put in a valid address ... 

Screen Shot 2021-12-10 at 3 47 54 PM

And then get an endless spinner ... 

Screen Shot 2021-12-10 at 3 48 01 PM

To end that or resubmit, I have to click 'back' and enter an addy

### Version

_No response_

### Relevant log output

_No response_

### LINKED ISSUE #591: [Bug]:
### Discord Discussion Link

https://discord.com/channels/808358975287722045/893564740427743292/919296334765391902

### What browsers are you seeing the problem on?

Chrome

### What were you trying to do?

- give bad input for view-only wallet address (intentionally)
- click 'Explore Tally' button
- dismiss the alert popup
- the button keeps the loading animation instead of flipping back to 'Explore Tally'

### What did not work?

It looks like the SharedButton remains in clicked state when it should not. (keeps showing the loading animation)

### Version

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by henryboldi**
Decimate janky alert modal.

Obliterate janky loader that didn't clear. (Closes #591)

👻

Closes #557

### CODE REVIEW

File: ui/pages/Onboarding/OnboardingViewOnlyWallet.tsx [hunk 1]
```diff
@@ -12,15 +13,10 @@ export default function OnboardingViewOnlyWallet(): ReactElement {
...
-  // Quick temp solution grabbed from
-  // https://ethereum.stackexchange.com/a/40670
-  function checkIfPlausibleETHAddress(checkAddress: string) {
-    return /^(0x){1}[0-9a-fA-F]{40}$/i.test(checkAddress)
-  }
+  const [errorMessage, setErrorMessage] = useState("")
 
   const handleSubmitViewOnlyAddress = useCallback(async () => {
-    if (checkIfPlausibleETHAddress(address)) {
+    if (isAddress(address)) {
```
Comment (mhluongo) [inline ui/pages/Onboarding/OnboardingViewOnlyWallet.tsx:19]: Can we slap on a quick `.trim()` here as well as when we `setCurrentAddress`? It's easy to copy an address in and make a mistake with a leading space

**Review by mhluongo — CHANGES_REQUESTED** [review]
Much improved, just want to get in one more fix before release so we can be done here.

File: ui/pages/Onboarding/OnboardingViewOnlyWallet.tsx (hunk 1 above)
Comment (henryboldi) [reply ui/pages/Onboarding/OnboardingViewOnlyWallet.tsx]: Absolutely! Added :)

**Review by mhluongo — APPROVED** [review]
:metal:

**Pull request merged**